src/gol.js

```diff
@@ -1,34 +1,36 @@
-export default function next(input) {
+export const countLiveNeighbours = (input, i, j) => {
+  let count = 0;
+
+  for (let x = -1; x < 2; x += 1) {
+    for (let y = -1; y < 2; y += 1) {
+      if (input[i + x] !== undefined && input[i + x][j + y] !== undefined && (x !== 0 || y !== 0)) {
+        count += input[i + x][j + y];
+      }
+    }
+  }
+
+  return count;
+};
+
+const next = (input) => {
   const output = input.map((row) => row.map((cell) => cell));
-  /*
-    input.map(function(row) {
-      return row.map(function(cell) {
-        return cell
-      })
-    })
-  */
 
   for (let i = 0; i < input.length; i += 1) {
     for (let j = 0; j < input[i].length; j += 1) {
       const liveNeighbours = countLiveNeighbours(input, i, j);
-      if (liveNeighbours < 2) {
-        output[i][j] = 0;
+      // Check if live cell should die
+      if (output[i][j] === 1) {
+        if (liveNeighbours < 2 || liveNeighbours > 3) {
```

Comment:
I like the simplification of these rules and the split for currently live cells and currently dead cells. 
Also the comment here is appreciated as this is basicaly the main business logic.